build.gradle

```diff
@@ -78,7 +78,21 @@ if (isWindows) {
   artifactClassifier = 'linux-' + osArch
 }
 
-String cmakeFlags = '--no-warn-unused-cli -D CMAKE_BUILD_TYPE=Release -D CMAKE_TOOLCHAIN_FILE=cmake/' + artifactClassifier + '.cmake'
+// Ensure we have a consistent combination of build config type and build directory
+// based on whether we want a debug build or not.
+// By default a "Release" build is performed.
+String cmakeFlags = '--no-warn-unused-cli -D CMAKE_TOOLCHAIN_FILE=cmake/' + artifactClassifier + '.cmake'
+project.ext.cmake_build_dir = "cmake-build-release"
+project.ext.cmake_build_type = "Release"
+if (mlDebug == "1")
+{
+  cmakeFlags = cmakeFlags + ' -D CMAKE_BUILD_TYPE=Debug'
+  project.ext.cmake_build_dir = "cmake-build-debug"
+  project.ext.cmake_build_type = "Debug"
+} else {
+  cmakeFlags = cmakeFlags + ' -D CMAKE_BUILD_TYPE=Release'
+}
```

Comment:
I would prefer not to duplicate as much logic between the CMake files and build.gradle.  It's just more to remember to update if we ever change anything in the future.

The CMake files already choose the build type based on the value of `ML_DEBUG`, so why not leave that aspect up to the CMake files.

It seems all we really need to do here is select the correct CMake build directory and then should leave the rest up to the existing logic in the CMake files.

Also the `mlDebug == 1` condition is not ideal, as now that variable is being interpreted in 3 different ways in the CMake system, shell scripts and in this file.  CMake will treat `on`, `yes`, `true`, `y` or any non-zero number the same whereas this only accepts `1`.  Shell scripts treat anything with non-zero length as true.  I think this should match one or other of those rather than introducing a 3rd interpretation.  (Ideally all 3 would interpret it the same.)

Also, we need similar logic in `docker_entrypoint.sh` - that should now leave the build type up to the CMake system to decide.  For `docker_entrypoint.sh` I think it would also be fine to use `cmake-build-docker` as a single CMake build directory instead of trying to determine whether to use release or debug.  The container is entirely throwaway, so it's not like Gradle where somebody may want to share cache variables between CLion and Gradle for example.